server/test/models/DFSPEndpointModelTest.js

```diff
@@ -0,0 +1,221 @@
+        console.log(`creating dfsp record=${dfsp.dfsp_id}`);
+        const dfspCreateResult = await DFSPModel.create(dfsp);
+        console.log(`dfspCreateResult=${dfspCreateResult}`);
+      }
+    });
+
+    afterEach('cleanup', async () => {
+      // Cleanup
+
+      console.log(`deleting dfsp records`);
+      for (const dfsp of dfspData) {
+        console.log(`deleting dfsp record=${dfsp.dfsp_id}`);
+        await DFSPModel.delete(dfsp.dfsp_id);
+      }
+      console.log(`deleting endpointList=${JSON.stringify(endpointIdList)}`);
+      for (const endpointId of endpointIdList) {
+        console.log(`deleting endpoint record=${endpointId}`);
+        await DFSPEndpointModel.delete(endpointId);
+      }
+    });
+
+    it('should create a valid Endpoint configuration', async () => {
+      // Act
+
+      // DFSPEndpointModel.create = async (id, state, type, direction, value) => {
+      endpointData.id = await DFSPEndpointModel.create(endpointData.dfsp_id, endpointData.state, endpointData.type, endpointData.direction, endpointData.value);
+      endpointIdList.push(endpointData.id);
+      console.log(`endpointCreateResult=${endpointData.id}`);
+
+      const findResult = await DFSPEndpointModel.findObjectById(endpointData.id);
+
+      // Assert
+      assert.equal(findResult.id, endpointData.id);
+      assert.equal(findResult.dfsp_id, endpointData.dfsp_id);
+      assert.equal(findResult.direction, endpointData.direction);
+      assert.equal(findResult.state, endpointData.state);
+      assert.equal(findResult.type, endpointData.type);
+      assert.equal(findResult.created_by, null);
+      assert.deepEqual(findResult.IPList, endpointData.value.IPList);
+      assert.beforeOrEqualDate(new Date(findResult.created_date), new Date());
+    });
+
+    it('should return latest Endpoint config record from findLastestObjectByDirection ordered by dfsp_endpoint.id', async () => {
+      // Act
+
+      // DFSPEndpointModel.create = async (id, state, type, direction, value) => {
+      endpointData.id = await DFSPEndpointModel.create(endpointData.dfsp_id, endpointData.state, endpointData.type, endpointData.direction, endpointData.value);
+      endpointIdList.push(endpointData.id);
+      console.log(`endpointCreateResult.1=${endpointData.id}`);
+      endpointData.id = await DFSPEndpointModel.create(endpointData.dfsp_id, endpointData.state, endpointData.type, endpointData.direction, endpointData.value);
+      endpointIdList.push(endpointData.id);
+      console.log(`endpointCreateResult.2=${endpointData.id}`);
+      endpointData.id = await DFSPEndpointModel.create(endpointData.dfsp_id, endpointData.state, endpointData.type, endpointData.direction, endpointData.value);
+      endpointIdList.push(endpointData.id);
+      console.log(`endpointCreateResult.3=${endpointData.id}`);
+
+      const findLastestObjectByDirectionResult = await DFSPEndpointModel.findLastestObjectByDirection(endpointData.dfsp_id, DirectionEnum.EGRESS);
```

Comment:
I think it's not quite clear what is the "latest object" is. Could we use just "result" or "endpoint" as a variable name